lib/BridgeCard.php

```diff
@@ -0,0 +1,259 @@
+<?php
+final class BridgeCard {
+
+	private static function buildFormatButtons($formats) {
+		$buttons = '';
+
+		foreach($formats as $name) {
+			$buttons .= '<button type="submit" name="format" value="'
+			. $name
+			. '">'
+			. $name
+			. '</button>'
+			. PHP_EOL;
+		}
+
+		return $buttons;
+	}
+
+	private static function getFormHeader($bridgeName, $isWarning = false) {
+		$form = <<<EOD
+			<form method="GET" action="?">
+				<input type="hidden" name="action" value="display" />
+				<input type="hidden" name="bridge" value="{$bridgeName}" />
+EOD;
+
+		if($isWarning) {
+			$form .= '<div class="secure-warning">Warning :
+This bridge is not fetching its content through a secure connection</div>';
+		}
+
+		return $form;
+	}
+
+	private static function getForm($bridgeName,
+	$formats,
+	$isActive = false,
+	$isWarning = false,
+	$parameterName = '',
+	$parameter = array()) {
+		$form = BridgeCard::getFormHeader($bridgeName, $isWarning);
+
+		foreach($parameter as $id => $inputEntry) {
+			if(!isset($inputEntry['exampleValue']))
+				$inputEntry['exampleValue'] = '';
+
+			if(!isset($inputEntry['defaultValue']))
+				$inputEntry['defaultValue'] = '';
+
+			$idArg = 'arg-'
+				. urlencode($bridgeName)
+				. '-'
+				. urlencode($parameterName)
+				. '-'
+				. urlencode($id);
+
+			$form .= '<label for="'
+				. $idArg
+				. '">'
+				. $inputEntry['name']
+				. ' : </label>'
+				. PHP_EOL;
+
+			if(!isset($inputEntry['type']) || $inputEntry['type'] === 'text') {
+				$form .= BridgeCard::getTextInput($inputEntry, $idArg, $id);
+			} elseif($inputEntry['type'] === 'number') {
+				$form .= BridgeCard::getNumberInput($inputEntry, $idArg, $id);
+			} else if($inputEntry['type'] === 'list') {
+				$form .= BridgeCard::getListInput($inputEntry, $idArg, $id);
+			} elseif($inputEntry['type'] === 'checkbox') {
+				$form .= BridgeCard::getCheckboxInput($inputEntry, $idArg, $id);
+			}
+		}
+
+		if($isActive) {
+			$form .= BridgeCard::buildFormatButtons($formats);
+		} else {
+			$form .= '<span style="font-weight: bold;">Inactive</span>';
+		}
+
+		return $form . '</form>' . PHP_EOL;
+	}
+
+	private static function getInputAttributes($entry) {
+		$retVal = '';
+
+		if(isset($entry['required']) && $entry['required'] === true)
+			$retVal .= ' required';
+
+		if(isset($entry['pattern']))
+			$retVal .= ' pattern="' . $entry['pattern'] . '"';
+
+		if(isset($entry['title']))
+			$retVal .= ' title="' . $entry['title'] . '"';
+
+		return $retVal;
+	}
+
+	private static function getTextInput($entry, $id, $name) {
+		return '<input '
+		. BridgeCard::getInputAttributes($entry)
+		. ' id="'
+		. $id
+		. '" type="text" value="'
+		. $entry['defaultValue']
```

Comment:
I am not sure here but is there any sanitation done before ?

Reply:
Yes, sort of. From line 45 onwards the blanks are filled in:

```PHP
if(!isset($inputEntry['exampleValue']))
	$inputEntry['exampleValue'] = '';

if(!isset($inputEntry['defaultValue']))
	$inputEntry['defaultValue'] = '';
```

Now that I think about it, [filter_var](http://php.net/manual/en/function.filter-var.php) is a better option though...